### PR TITLE
Make sure that 'debug-sites' directory is created before being accessed.

### PR DESCRIPTION
**Pull request by slaufmann**
This pull request fixes an issue i stumbled across:

Even though the 'debug-sites' directory is present in a checked out version of this
software it may be deleted by a user in order to clean up the debug logs.
Therefore the directory is removed from version-control and created by the software
if not already present before the first access.

### CODE REVIEW

File: crawler.py [hunk 1]
```diff
@@ -332,8 +332,12 @@ def truncate(string, max_len):
 
 
 def debug_dump_site_html(name, html):
-    with open(f"debug-sites/site-{name}.html", "w") as test_log:
-        print(html, file=test_log)
+    dump_sites_path = Path("debug-sites")
+    dump_sites_path.mkdir(parents=True, exist_ok=True)
+    site_name = "site-{}.html".format(name)
+    site_debug_path = dump_sites_path / f"sites-{name}.html"
```
Comment (grandchild) [inline crawler.py]: I don't get this pair of lines. Is the first (`site_name = ...`) a leftover?

**Review by grandchild — CHANGES_REQUESTED** [review]
One leftover line (I think), but otherwise very nice!

**Comment by grandchild** [comment]
Oh, and one minor suggestion for the commit message: Forcing yourself to use active voice often makes the message considerably more concise:

> Make sure that 'debug-sites' directory **is created** before **being accessed**.

:arrow_down:

> Ensure creation of 'debug-sites' directory before access

File: crawler.py (hunk 1 above)
Comment (slaufmann) [reply crawler.py]: I think you are right. This line is obsolete then.

**Comment by slaufmann** [comment]
I think the branch is ready for merging now:
- removed obsolete line
- updated commit message
- rebased onto last commit in main

**Comment by grandchild** [comment]
Thanks so much! :)

**Pull request merged**